File: Advanced Methods of Programming/Toy Language Interpreter/src/Model/Statements/If.java
package Model.Statements;

import Model.Expressions.IExpression;
import Model.ProgramState;

public class If implements IStatement {
    private IExpression exp;
    private IStatement thenS;
    private IStatement elseS;

    public If(IExpression e, IStatement t){
        exp = e;
        thenS = t;
        elseS = null;
    }
    public If(IExpression e, IStatement t, IStatement el){
        exp = e;
        thenS = t;
        elseS = el;
    }

    @Override
    public ProgramState exec(ProgramState state){
        if (exp.eval(state) > 0){
            return thenS.exec(state);
        }
        if (elseS == null){
            return null;
        }
        return elseS.exec(state);
    }

    @Override
    public String toString() {
        return "IF(" + exp.toString() + ") THEN (" + thenS.toString() + " ) ELSE (" + elseS.toString() + ")";
    }

}

File: Advanced Methods of Programming/CSharp Toy Language Interpreter/ToyLanguageInterpreter/Src/Model/Statements/If.cs
using ToyLanguageInterpreter.Src.Model.Expressions;

namespace ToyLanguageInterpreter.Src.Model.Statements
{
    class If : IStatement
    {
        private IExpression exp;
        private IStatement thenS;
        private IStatement elseS;

        public If(IExpression e, IStatement t)
        {
            exp = e;
            thenS = t;
            elseS = null;
        }

        public If(IExpression e, IStatement t, IStatement el)
        {
            exp = e;
            thenS = t;
            elseS = el;
        }

        public ProgramState Exec(ProgramState state)
        {
            if (exp.Eval(state) > 0)
            {
                return thenS.Exec(state);
            }

            if (elseS == null)
            {
                return null;
            }

            return elseS.Exec(state);
        }
        
    
        public override string ToString()
        {
            return "IF(" + exp.ToString() + ") THEN (" + thenS.ToString() + " ) ELSE (" + elseS.ToString() + ")";
        }
    }
}
